app/models/assets/import.rb

```diff
@@ -309,19 +309,11 @@ def import_remote_asset(remote_asset, barcode, import_step)
       Asset
         .create!(barcode: barcode, uuid: remote_asset.uuid, facts: [])
         .tap do |asset|
-          FactChanges
-            .new
-            .tap do |updates|
-              updates.replace_remote(asset, 'a', sequencescape_type_for_asset(remote_asset))
-              updates.replace_remote(asset, 'remoteAsset', remote_asset.uuid)
-            end
-            .apply(import_step)
-
           # We initialize the asset with an empty facts array, but then modify it indirectly
           # Ideally this wouldn't be the case, and we'd keep the instance in sync. However
           # for now we unload the association to ensure we don't end up working with stale data
           asset.facts.reset
-          asset.refresh_from_remote(remote_asset: remote_asset)
```

Comment:
Pass in the import step to associate the facts with that as well. As an added benefit our import step now gets properly resolved.